### PR TITLE
⬆️ Upgrade to Flow 0.111.1

### PR DESCRIPTION
**Pull request by dmnd**
I made about 3 fixes then decided to just suppress everything else. The new errors aren't trivial to work around.

### CODE REVIEW

File: InlineEdit.jsx [hunk 1]
```diff
@@ -250,14 +250,17 @@ function InlineEdit<T>({
...
+
   return (
-    <div
-      className={css(styles.container)}
-      style={{
-        ...(transitioningIn ? transitionInStyle : {}),
-        ...(transitioningOut ? transitionOutStyle : {}),
-      }}
-    >
+    <div className={css(styles.container)} style={style}>
```
Comment (dmnd) [inline InlineEdit.jsx]: manual fix

File: TextInput.jsx [hunk 2]
```diff
@@ -142,7 +142,7 @@ function Sigil({
     <span
       className={css(styles.sigil)}
       style={{
-        [side]: `${paddingDefaultTextInput}px`,
+        [(side: string)]: `${paddingDefaultTextInput}px`,
```
Comment (dmnd) [inline TextInput.jsx]: manual fix

File: flow-typed-overrides/npm/react-popper_v1.3.x.js [hunk 3]
```diff
@@ -28,8 +28,8 @@ declare module 'react-popper' {
   declare type Style = {[string]: string | number};
   declare type getRefFn = (?HTMLElement) => void;
   declare type ReferenceElement = ReferenceObject | HTMLElement | null;
-  declare type StyleOffsets = { top: number, left: number };
-  declare type StylePosition = { position: 'absolute' | 'fixed' };
+  declare type StyleOffsets = {| top: number, left: number |};
+  declare type StylePosition = {| position: 'absolute' | 'fixed' |};
```
Comment (dmnd) [inline flow-typed-overrides/npm/react-popper_v1.3.x.js]: manual fix

File: select/DropdownList.jsx [hunk 4]
```diff
@@ -72,7 +72,9 @@ export default function DropdownList({
           label={option.label}
           customView={option.customView}
           isHighlighted={option.label === highlightedOption}
+          // $FlowUpgradeFixMe(0.110.1 -> 0.111.1)
```
Comment (kejistan) [inline select/DropdownList.jsx]: what are the errors here? This seems different from the other spread-related fixmes.

**Comment by kejistan** [comment]
Also: 0.111.2 is available, I doubt it changes anything for this PR (in terms of suppressions).

File: select/DropdownList.jsx (hunk 4 above)
Comment (dmnd) [reply select/DropdownList.jsx]: Good catch. These were sketchy null checks. I'll remove the suppressions

File: select/DropdownList.jsx [hunk 5]
```diff
@@ -72,10 +72,8 @@ export default function DropdownList({
           label={option.label}
           customView={option.customView}
           isHighlighted={option.label === highlightedOption}
-          // $FlowUpgradeFixMe(0.110.1 -> 0.111.1)
-          onClick={option.disabled ? () => {} : onClick}
-          // $FlowUpgradeFixMe(0.110.1 -> 0.111.1)
-          disabled={option.disabled}
+          onClick={option.disabled === true ? () => {} : onClick}
+          disabled={option.disabled === true}
```
Comment (kejistan) [inline select/DropdownList.jsx]: I think you can keep this as `disabled={option.disabled}` if you update the prop type on `DropdownOption` to be optional (it is optional, the type is just wrong).

Comment (dmnd) [reply select/DropdownList.jsx]: Good idea, done

**Comment by dmnd** [comment]
> Also: 0.111.2 is available, I doubt it changes anything for this PR (in terms of suppressions).

[not yet](https://github.com/flowtype/flow-bin/releases)

**Pull request merged**